feat: add profanity filter

AmanNegi opened the pull request:
## Issue Description

Block users from using profane words, using LLM to classify if a text contains profanity or not. Validate user input when:
- user passes a query for prompt generation.
- user passes a prompt for GIF generation.

> Fixes: #9 

## Preview

![profanity-img](https://github.com/RocketChat/Apps.AI.GIF/assets/37607224/72f18561-0814-4182-946f-f4d978220426)

AmanNegi committed:
feat: add profanity filter

- Block users from using profane words, using LLM to classify if a text
  contains profanity or not.

Nabhag8848 commented:
@AmanNegi this doesn't include readme changes + llm url to set from settings as discussed.

AmanNegi committed:
docs: update readme

- update features progress.
- add banner as well as architecture image.
- update setup steps

AmanNegi committed:
chore: remove heading from gif message title

AmanNegi commented:
@Nabhag8848 I have made the requested changes could you look into this.

AmanNegi committed:
chore: update model url

Nabhag8848 reviewed (approved):
LGTM 🔥
